### PR TITLE
Add documentation, adapt the code a bit

### PR DESCRIPTION
**Pull request by mardiros**
Add a bit of documentation here

### CODE REVIEW

File: docs/source/users/oauth2_settings.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+| OAUTH2_LOGOUT_URL         | url to redirect on logout             | yes       | str       |
++---------------------------+---------------------------------------+-----------+-----------+
+| OAUTH2_TIMEOUT            | HTTP Timeout in seconds               | no (30)   | int       |
++---------------------------+---------------------------------------+-----------+-----------+
+| OAUTH2_VERIFY_CERTIFICATE | Check TLS while consuming tokens      | no (True) | bool      |
++---------------------------+---------------------------------------+-----------+-----------+
+
+
+The settings OAUTH2_LOAD_USERINFO is a function that take an access_token in parameter,
+and build a python dict or raise a PermissionDenied error.
```
Comment (yanndinendal) [inline docs/source/users/oauth2_settings.rst]:
```suggestion
The settings `OAUTH2_LOAD_USERINFO` is a function that takes an `access_token` in parameter,
and builds a python dict or raises a `PermissionDenied` error.
```

File: docs/source/users/oauth2_settings.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+Basically, this method is about fetching some information on the user loaded using
+OAuth2.0 API and decide to grant the user to logged in, and to get the role of 
+that user.
```
Comment (yanndinendal) [inline docs/source/users/oauth2_settings.rst]:
```suggestion
Basically, this method is about fetching some information on the user loaded using
OAuth2.0 API and deciding to grant the user to log in, and to get the role of 
that user.
```

File: docs/source/users/oauth2_settings.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+The userinfo dict contains the following keys:
+
++--------------+------------------------------------+--------------------------------+-----------+
+| key          | description                        | mandatory (default)            | type      |
++==============+====================================+================================+===========+
+| username     | user identifier                    | yes                            | str       |
++--------------+------------------------------------+--------------------------------+-----------+
+| is_superuser | make the user an admin on creation | yes                            | bool      |
```
Comment (yanndinendal) [inline docs/source/users/oauth2_settings.rst]:
```suggestion
| is_superuser | makes the user an admin on creation | yes                            | bool      |
```

File: src/wagtail_oauth2/views.py [hunk 4]
```diff
@@ -1,8 +1,7 @@
 """
-Views to login using Gandi ID.
+Views to login using an OAuth2.0 Authorization Server.
```
Comment (yanndinendal) [inline src/wagtail_oauth2/views.py]:
```suggestion
Views to log in using an OAuth2.0 Authorization Server.
```

File: docs/source/users/oauth2_settings.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
++===========================+=======================================+===========+===========+
+| OAUTH2_CLIENT_ID          | OAuth2.0 client id                    | yes       | str       |
++---------------------------+---------------------------------------+-----------+-----------+
+| OAUTH2_CLIENT_SECRET      | OAuth2.0 client secret                | yes       | str       |
++---------------------------+---------------------------------------+-----------+-----------+
+| OAUTH2_AUTH_URL           | /authorize endpoint                   | yes       | str       |
++---------------------------+---------------------------------------+-----------+-----------+
+| OAUTH2_TOKEN_URL          | /token url                            | yes       | str       |
++---------------------------+---------------------------------------+-----------+-----------+
+| OAUTH2_LOAD_USERINFO      | Load user info from the oauth2 server | yes       | callable  |
```
Comment (yanndinendal) [inline docs/source/users/oauth2_settings.rst:20]: Do you have an example?

**Review by yanndinendal — COMMENTED** [review]
:+1:

File: docs/source/users/oauth2_settings.rst (hunk 5 above)
Comment (mardiros) [reply docs/source/users/oauth2_settings.rst:20]: DONE

**Pull request merged**